packages/repository/src/repositories/relation.factory.ts

```diff
@@ -35,21 +40,76 @@ export function createHasManyRepositoryFactory<
   ForeignKeyType
 >(
   relationMetadata: HasManyDefinition,
-  targetRepository: EntityCrudRepository<Target, TargetID>,
+  targetRepositoryGetter: Getter<EntityCrudRepository<Target, TargetID>>,
 ): HasManyRepositoryFactory<Target, ForeignKeyType> {
+  const meta = resolveHasManyMetadata(relationMetadata);
+  debug('Resolved relation metadata: %o', meta);
   return function(fkValue: ForeignKeyType) {
-    const fkName = relationMetadata.keyTo;
-    if (!fkName) {
-      throw new Error(
-        'The foreign key property name (keyTo) must be specified',
-      );
-    }
     // tslint:disable-next-line:no-any
-    const constraint: any = {[fkName]: fkValue};
+    const constraint: any = {[meta.keyTo]: fkValue};
     return new DefaultHasManyEntityCrudRepository<
       Target,
       TargetID,
       EntityCrudRepository<Target, TargetID>
-    >(targetRepository, constraint as DataObject<Target>);
+    >(targetRepositoryGetter, constraint as DataObject<Target>);
   };
 }
+
+export type HasManyResolvedDefinition = HasManyDefinition & {keyTo: string};
+/**
+ * Resolves given hasMany metadata if target is specified to be a resolver.
+ * Mainly used to infer what the `keyTo` property should be from the target's
+ * belongsTo metadata
+ * @param relationMeta hasMany metadata to resolve
+ */
+export function resolveHasManyMetadata(
+  relationMeta: HasManyDefinition,
+): HasManyResolvedDefinition {
+  if (typeof relationMeta.target !== 'function') {
+    const reason = 'target model is required.';
```

Comment:
I might be mistaken, but isn't this check similar to the one below?

Reply:
Good catch 👍